lib/6to5/generation/buffer.js

```diff
@@ -67,32 +67,57 @@ Buffer.prototype.removeLast = function (cha) {
 
 Buffer.prototype.newline = function (i, removeLast) {
   if (this.format.compact) return;
-
-  if (_.isBoolean(i)) {
-    removeLast = i;
-    i = null;
-  }
+  removeLast = removeLast || false;
 
   if (_.isNumber(i)) {
     if (this.endsWith("{\n")) i--;
     if (this.endsWith(util.repeat(i, "\n"))) return;
 
-    for (var j = 0; j < i; j++) {
-      this.newline(null, removeLast);
+    while (i--) {
+      this._newline(removeLast);
     }
     return;
   }
 
+  if (_.isBoolean(i)) {
+    removeLast = i;
+  }
+
+  this._newline(removeLast);
+};
+
+Buffer.prototype._newline = function (removeLast) {
   if (removeLast && this.isLast("\n")) this.removeLast("\n");
 
   this.removeLast(" ");
 
   // remove whitespace if last character was a newline
-  this.buf = this.buf.replace(/\n +$/, "\n");
-
+  this._removeSpacesAfterLastNewline();
   this._push("\n");
 };
 
+/**
+ * If buffer ends with a newline and some spaces after it, trim those spaces.
+ */
+Buffer.prototype._removeSpacesAfterLastNewline = function () {
+  var lastNewlineIndex = this.buf.lastIndexOf('\n');
```

Comment:
Potentially. I'll probably just end up adding an option to nerf the pretty code generator output for those that want it or make the code generator disable the pretty output if the file exceeds a certain size, what do you think?